error.go

```diff
@@ -72,22 +65,242 @@ func collectMessages(data interface{}, messages []string) []string {
 	return messages
 }
 
-type apiErrorResponse struct {
-	Error *APIError `json:"error,omitempty"`
+// Error provides a pretty printed string of an Error object based on present data.
+func (e *Error) Error() string {
+	if e.Message != "" {
+		if e.Code != "" {
+			return e.Code + " " + e.Message.(string)
+		}
+		return e.Message.(string)
+	}
+	return e.Code
+}
+
+// LibraryError is the base type for all errors/exceptions in this EasyPost library.
+type LibraryError struct {
+	// Message is a human-readable error description.
+	Message string
+}
+
+// Error provides a pretty printed string of a LibraryError object.
+func (e *LibraryError) Error() string {
+	return e.Message
+}
+
+// Local error types
+
+type LocalError struct {
+	LibraryError
+}
+
+type ValidationError struct {
+	LocalError
+}
+
+var EndOfPaginationError = &LocalError{LibraryError{Message: NoPagesLeftToRetrieve}}
+
+type FilteringError struct {
+	LocalError
+}
+
+type InvalidObjectError struct {
+	ValidationError
+}
+
+type InvalidParameterError struct {
+	ValidationError
+}
+
+type JsonError struct {
+	LocalError
+}
+
+type JsonDeserializationError struct {
+	JsonError
+}
+
+type JsonSerializationError struct {
+	JsonError
+}
+
+type JsonNoDataError struct {
+	JsonError
+}
+
+type MissingParameterError struct {
+	LocalError
+}
+
```

Comment:
thought: not a fan of the cascading errors of doom here. The multiple nested concept is difficult to follow.

Reply:
Resolved